woocommerce/woocommerce-android
# Shipping Labels: view order summary and Woo discount p1

First part of #3617, this just adds the layout of the order summary, and the logic to display it. But as #3525 is not done yet, we can't have a real calculation of the order price and discount.

|  |  |
| --- | --- |
|![Screenshot_1614175365](https://user-images.githubusercontent.com/1657201/109011652-6aeae500-76b1-11eb-9f04-67f60d9c714d.png)|![Screenshot_1614175367](https://user-images.githubusercontent.com/1657201/109011663-6de5d580-76b1-11eb-9375-c7c65396bea8.png)|
|![Screenshot_1614176182](https://user-images.githubusercontent.com/1657201/109013324-4bed5280-76b3-11eb-8f2a-733abc10bf4a.png)| |


#### Testing
1. Change this [line](https://github.com/woocommerce/woocommerce-android/blob/eb824a2d966f4f20160fd35099e92485e5551fba/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt#L157) to: 
``` kotlin
      is SideEffect.ShowCarrierOptions -> handleResult { Event.ShippingCarrierSelected }
```
to allow completing the shipping carriers step.
2. Open details of an order that satisfies the shipping label creation.
3. Click on "create shipping label"
4. Confirm that the order summary is hidden.
5. Pass all steps.
6. Confirm that the order summary is displayed after reaching the payment step.
7. Click on the ℹ️ image, and confirm that the discount bottom sheet is displayed.

To test the other scenario: order without discount, use `BigDecimal.ZERO` for the discount field here: https://github.com/woocommerce/woocommerce-android/blob/a9b4af2f1aa93b37c340a7d591e0861bee928887/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/shippinglabels/creation/CreateShippingLabelViewModel.kt#L288-L290

Update release notes:
